### PR TITLE
use permit for tendertokens in swap

### PR DESCRIPTION
**Pull request by vujevits**

### CODE REVIEW

File: packages/app/src/components/approve/ApproveToken.tsx [hunk 1]
```diff
@@ -1,28 +1,55 @@
...
+          spender,
+          amount?.toString(),
+          tenderTokenApproval.deadline,
+          tenderTokenApproval.v,
+          tenderTokenApproval.r,
+          tenderTokenApproval.s
+        );
+      } else {
+        await approveToken(spender, constants.MaxUint256);
+      }
```
Comment (kyriediculous) [inline packages/app/src/components/approve/ApproveToken.tsx:52]: This is not how permit works. This makes three steps in the UX of what is now two steps in the UX. 

1. Sign the message
2. Send permit transaction
3. Make whatever call

Just like in the [add liquidity script example](https://github.com/Tenderize/tender-core/blob/master/scripts/add_liquidity.ts) you should use multicall

**Review by kyriediculous — CHANGES_REQUESTED** [review]

File: packages/app/src/utils/tenderSwapHooks.ts [hunk 2]
```diff
@@ -65,6 +66,39 @@ export const useCalculateSwap = (pool: string, tokenFrom: string, amount: BigNum
...
+  const { state, send: multicall } = useContractFunction(contracts[protocolName].tenderSwap, "multicall");
+  const { library } = useEthers();
+
+  const swapWithPermit = async (
+    tokenAddress: string,
+    tokenAmount: BigNumber,
+    minAmount: BigNumber,
+    deadline: number
+  ) => {
+    const permit = await signERC2612Permit(library, token, owner ?? "", spender, amount?.toString(), getDeadline());
```
Comment (kyriediculous) [inline packages/app/src/utils/tenderSwapHooks.ts:85]: Library is not a thing here

File: packages/app/src/components/approve/PermitToken.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  amount: BigNumber;
+  owner: string | undefined | null;
+};
+
+const PermitToken: FC<Props> = ({ symbol, spender, show, token, owner, amount }) => {
+  const { state: approveTx, send: permitToken } = useContractFunction(token, "permit", {
+    transactionName: `Permit ${symbol}`,
+  });
+
+  const { library } = useEthers();
```
Comment (kyriediculous) [inline packages/app/src/components/approve/PermitToken.tsx:25]: remove

File: packages/app/src/components/approve/PermitToken.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      );
+      await permitToken(
+        owner,
+        spender,
+        amount?.toString(),
+        tenderTokenApproval.deadline,
+        tenderTokenApproval.v,
+        tenderTokenApproval.r,
+        tenderTokenApproval.s
+      );
```
Comment (kyriediculous) [inline packages/app/src/components/approve/PermitToken.tsx:47]: remove

File: packages/app/src/utils/tenderSwapHooks.ts (hunk 2 above)
Comment (vujevits) [reply packages/app/src/utils/tenderSwapHooks.ts:85]: image

File: packages/app/src/components/approve/PermitToken.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                  round: {
+                    size: "20px",
+                  },
+                  background: "rgba(0,0,0,0.4)",
+                  elevation: "none",
+                }}
+                content={
+                  <Box width="medium" elevation="none" pad="medium">
+                    <Text color="white">
+                      {`You must give the Tenderize smart contracts permission to use your ${symbol}. This will not cost you anything.`}
```
Comment (kyriediculous) [inline packages/app/src/components/approve/PermitToken.tsx:80]: Remove , this is untrue.

File: packages/app/src/components/approve/PermitToken.tsx (hunk 3 above)
Comment (vujevits) [reply packages/app/src/components/approve/PermitToken.tsx:25]: this component is unused

File: packages/app/src/components/approve/ApproveToken.tsx [hunk 6]
```diff
@@ -41,7 +40,7 @@ const ApproveToken: FC<Props> = ({ symbol, spender, show, token }) => {
         label={
           <>
             <Box justify="center" align="center" direction="row" gap="small" pad={{ horizontal: "xsmall" }}>
-              {isPendingTransaction(approveTx) && <LoadingButtonContent />}
+              {approveTx.status === "Mining" && <LoadingButtonContent />}
```
Comment (kyriediculous) [inline packages/app/src/components/approve/ApproveToken.tsx:43]: This change needs to undone

**Review by kyriediculous — CHANGES_REQUESTED** [review]

**Review by kyriediculous — APPROVED** [review]
LGTM 🚢

**Pull request merged**